### PR TITLE
Fix resolved view decorator

### PR DESCRIPTION
**Pull request by bas080**

### CODE REVIEW

File: dist/amd/decorators/resolved-view.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ define(['exports', 'aurelia-dependency-injection', './../view-manager'], functio
   function resolvedView(namespace, view) {
     return function resolvedViewDecorator(target) {
       var viewManager = _aureliaDependencyInjection.Container.instance.get(_viewManager.ViewManager);
-      target.getViewStrategy = viewManager.resolve(namespace, view);
+
+      target.prototype.getViewStrategy = viewManager.resolve.bind(viewManager, namespace, view);
+
+      (0, _aureliaTemplating.useView)(viewManager.resolve(namespace, view))(target);
```
Comment (VMBindraban) [inline dist/amd/decorators/resolved-view.js]: Wut?

File: src/decorators/resolved-view.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
 import {Container} from 'aurelia-dependency-injection';
 import {ViewManager} from './../view-manager';
+import {useView} from 'aurelia-templating';
 
 export function resolvedView(namespace, view) {
   return function resolvedViewDecorator(target) {
     let viewManager        = Container.instance.get(ViewManager);
```
Comment (VMBindraban) [inline src/decorators/resolved-view.js:7]: Alignment

File: src/decorators/resolved-view.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
-    target.getViewStrategy = viewManager.resolve(namespace, view);
 
-    return target; /* handy for testing */
+    target.prototype.getViewStrategy = viewManager.resolve.bind(viewManager, namespace, view);
+
+    useView(viewManager.resolve(namespace, view))(target);
```
Comment (RWOverdijk) [inline src/decorators/resolved-view.js:11]: Why not simply do:

``` js
import {Container} from 'aurelia-dependency-injection';
import {ViewManager} from './../view-manager';
import {RelativeViewStrategy, useViewStrategy} from 'aurelia-templating';

export function resolvedView(namespace, view) {
  let viewManager = Container.instance.get(ViewManager);

  return useViewStrategy(new RelativeViewStrategy(viewManager.resolve(namespace, view)));
}
```

The only thing I'm not sure about in my example, is if RelativeViewStrategy is actually exported from `'aurelia-templating'` or `'aurelia-templating/view-strategy'`.

**Comment by bas080** [comment]
Last time I checked it was not. It is used internally in aurelia template ng.

I do agree that decorators are not meant to be used in this manner. It does work just fine

**Comment by RWOverdijk** [comment]
@bas080 I know for a fact that it's being exported. https://github.com/aurelia/templating/blob/master/src/view-strategy.js#L44

My question was, which path is exporting it. I know now that my above example _should_ work. https://github.com/aurelia/templating/blob/master/dist/es2015/aurelia-templating.js#L387

**Pull request merged**